### DISCUSSION TITLE
PROJECTIONS_CHANGES_ENABLED disabled upsert but not strategies

### DISCUSSION THREAD

**Discussion by fabio-percivaldi** [mia-platform/community#101] [Ideas]
i tried to do a check on the codebase and i'm 70% sure that if we put PROJECTIONS_CHANGES_ENABLED to false projections changes are not upserted, but strategies are still executed, if the reason of turning off projections changes is to enhance the performance of an initial load, strategies are useless and slow down the operation time. Even if the purpose of turning off this setting is to avoid the generation of projection changes at all (and not performance reason) we don't need to execute strategies. 

i see there is a TODO here https://git.tools.mia-platform.eu/platform/fast-data/libraries/kafka-importer-lib/-/blob/master/src/projections/MongoProjectionsChanges.js#L165

but i'm not sure is totally related to this discussion (or maybe yes)

**Comment by ianFar96** [comment]
Hi @fabio-percivaldi, you are probably right in your supposition, we will have a look at it to remove the strategy execution when PROJECTIONS_CHANGES_ENABLED is false. Thank you for your suggestion!

**Comment by danibix95** [comment]
Close because there have been made many different changes to the code base since this discussion has been opened. Furthermore, the combination of `projection-storer` and `single-view-trigger-generator` yields higher performances than the solely `real-time-updater`
